### PR TITLE
Switch nightlies to GitHub releases.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

Second half of https://github.com/netdata/netdata/pull/13961.

This switches to pulling nightly builds from https://github.com/netdata/netdata-nightlies instead of our GCS buckets when installing static builds or local builds.

##### Test Plan

Easily testable by simply attempting to use the kickstart script from this PR without using native packages and noting where it downloads the files from.

Equivalent testing can be done with the updater script.

##### Additional Information

We will still need changes to the update availability checks, but that will need to be in a different PR.

### CODE REVIEW

**Comment by Ferroin** [comment]
Rebased to resolve merge conflicts (and re-trigger CI with the repo this is trying to pull artifacts from now being public).

**Comment by andrewm4894** [comment]
very small one - will badge on readme need to be updated as part of this?

**Comment by Ferroin** [comment]
> very small one - will badge on readme need to be updated as part of this?

Nice catch, it will indeed need to be updated.

**Comment by Ferroin** [comment]
Rebased to fix invalid changes and remove testing code.

**Comment by Ferroin** [comment]
Closing and reopening to force GHA checks to run.

File: packaging/installer/netdata-updater.sh [hunk 1]
```diff
@@ -439,7 +442,10 @@ self_update() {
 
 parse_version() {
   r="${1}"
-  if echo "${r}" | grep -q '^v.*'; then
+  if [ "${r}" = "latest" ]; then
+    printf "99999999999999"
```
Comment (thiagoftsm) [inline packaging/installer/netdata-updater.sh:446]: Is there any special motive to use this `magic number`?

**Comment by thiagoftsm** [comment]
@Ferroin ,

After to test this PR I observed that it is finishing with the following warning:

```sh
Complete!
 OK 

 WARNING  Could not find netdata-updater.sh. This means that auto-updates cannot (currently) be enabled on this system. See https://learn.netdata.cloud/docs/agent/packaging/installer/update for more information about updating Netdata.

Successfully installed the Netdata Agent.

The following non-fatal warnings or errors were encountered:

  - Could not find netdata-updater.sh. This means that auto-updates cannot (currently) be enabled on this system. See https://learn.netdata.cloud/docs/agent/packaging/installer/update for more information about updating Netdata.

Official documentation can be found online at https://learn.netdata.cloud/docs/.

Looking to monitor all of your infrastructure with Netdata? Check out Netdata Cloud at https://app.netdata.cloud.

Join our community and connect with us on:
  - GitHub: https://github.com/netdata/netdata/discussions
  - Discord: https://discord.gg/5ygS846fR6
  - Our community forums: https://community.netdata.cloud/
```

Another thing I observed it is the fact that I did not have netdata started after to use `kickstart`, but I could start it using `systemctl`.

I ran all tests on `CentOS 7.9`

File: packaging/installer/netdata-updater.sh (hunk 1 above)
Comment (Ferroin) [reply packaging/installer/netdata-updater.sh:446]: It’s guaranteed to be larger than any other version number that the function might produce. I should probably add a comment to that effect though.

**Comment by Ferroin** [comment]
> @Ferroin ,
> 
> After to test this PR I observed that it is finishing with the following warning:
> 
> ```shell
> Complete!
>  OK 
> 
>  WARNING  Could not find netdata-updater.sh. This means that auto-updates cannot (currently) be enabled on this system. See https://learn.netdata.cloud/docs/agent/packaging/installer/update for more information about updating Netdata.
> 
> Successfully installed the Netdata Agent.
> 
> The following non-fatal warnings or errors were encountered:
> 
>   - Could not find netdata-updater.sh. This means that auto-updates cannot (currently) be enabled on this system. See https://learn.netdata.cloud/docs/agent/packaging/installer/update for more information about updating Netdata.
> 
> Official documentation can be found online at https://learn.netdata.cloud/docs/.
> 
> Looking to monitor all of your infrastructure with Netdata? Check out Netdata Cloud at https://app.netdata.cloud.
> 
> Join our community and connect with us on:
>   - GitHub: https://github.com/netdata/netdata/discussions
>   - Discord: https://discord.gg/5ygS846fR6
>   - Our community forums: https://community.netdata.cloud/
> ```
> 
> Another thing I observed it is the fact that I did not have netdata started after to use `kickstart`, but I could start it using `systemctl`.
> 
> I ran all tests on `CentOS 7.9`

Hmm, that’s kind of unexpected, as this should not be touching anything related to either point. Can you confirm whether or not you see the same behavior with the stock kickstart script?

File: packaging/installer/netdata-updater.sh (hunk 1 above)
Comment (thiagoftsm) [reply packaging/installer/netdata-updater.sh:446]: Please, add a comment to simplify our understanding.

**Comment by thiagoftsm** [comment]
> 

During my tests I used `./packaging/installer/kickstart.sh`, and I have this inside `/usr/libexec/netdata`:

```sh
[root@192 netdata]# ls
charts.d  edit-config  install-service.sh  plugins.d  python.d
[root@192 netdata]# pwd
``` 

when I compile our master branch I have this:

```sh
root@hades:/home/thiago/Netdata/netdata# ls /usr/libexec/netdata
charts.d/  install-service.sh*  netdata-uninstaller.sh*  netdata-updater.sh*  plugins.d/  python.d/
root@hades:/home/thiago/Netdata/netdata# 
```

**Comment by Ferroin** [comment]
> > 
> 
> During my tests I used `./packaging/installer/kickstart.sh`, and I have this inside `/usr/libexec/netdata`:
> 
> ```shell
> [root@192 netdata]# ls
> charts.d  edit-config  install-service.sh  plugins.d  python.d
> [root@192 netdata]# pwd
> ```
> 
> when I compile our master branch I have this:
> 
> ```shell
> root@hades:/home/thiago/Netdata/netdata# ls /usr/libexec/netdata
> charts.d/  install-service.sh*  netdata-uninstaller.sh*  netdata-updater.sh*  plugins.d/  python.d/
> root@hades:/home/thiago/Netdata/netdata# 
> ```

I’m unable to reproduce this on my end using either CentOS 7 or any newer RPM distro, at least when using a local build.

**Comment by thiagoftsm** [comment]
> > > 
> > 
> > 
> > During my tests I used `./packaging/installer/kickstart.sh`, and I have this inside `/usr/libexec/netdata`:
> > ```shell
> > [root@192 netdata]# ls
> > charts.d  edit-config  install-service.sh  plugins.d  python.d
> > [root@192 netdata]# pwd
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > when I compile our master branch I have this:
> > ```shell
> > root@hades:/home/thiago/Netdata/netdata# ls /usr/libexec/netdata
> > charts.d/  install-service.sh*  netdata-uninstaller.sh*  netdata-updater.sh*  plugins.d/  python.d/
> > root@hades:/home/thiago/Netdata/netdata# 
> > ```
> 
> I’m unable to reproduce this on my end using either CentOS 7 or any newer RPM distro, at least when using a local build.

I removed netdata completely to have these issues. I will try with another.

**Comment by thiagoftsm** [comment]
@Ferroin on Alma I did not have the reported issue, so I am considering this can be garbage on old CentOS VM.

**Review by thiagoftsm — APPROVED** [review]
As I explained [here](https://github.com/netdata/netdata/pull/14020#issuecomment-1379660423) it looks like I found a corner case, and considering that the other issue was explained, LGTM!

**Comment by MrZammler** [comment]
Hi @Ferroin 

Tested on a pi, running `Raspbian GNU/Linux 10 (buster)`. Appears to install ok (using --static-only), a couple of points:

* It installed version `v1.37.0-93-nightly` which is the latest, I assume no new builds yet, just thought to point it.

* I got: 
```
 --- Install (but not enable) netdata updater tool --- 
cat: /system/netdata-updater.timer: No such file or directory
cat: /system/netdata-updater.service: No such file or directory
Update script is located at /opt/netdata/usr/libexec/netdata/netdata-updater.sh
```

Yet, the script seems installed ok under `/etc/cron.daily`. Perhaps not related to this PR.

**Comment by MrZammler** [comment]
`--static-only` on Ubuntu 22.04 ok.

One note here, we need to maybe check: I had a package installed there, which I uninstalled using the uninstall script, yet it deleted the files rather than using the package manager to uninstall it. Is this expected?

**Comment by MrZammler** [comment]
Centos 7 Core, static build ok.
Centos 7 core, local build ok (and I get ok the files mentioned above under `/usr/libexec/netdata/`)

**Comment by MrZammler** [comment]
@Ferroin should we get a new nightly version up there, and also check the updater from this PR? Running it now identifies correctly that we are on the latest (-93) do you want to test it?

**Comment by Ferroin** [comment]
> @Ferroin should we get a new nightly version up there, and also check the updater from this PR? Running it now identifies correctly that we are on the latest (-93) do you want to test it?

That’s probably a good idea, though it means installing now with auto-updates disabled, waiting 24 hours, and then running the updater script from this PR by hand with `--no-updater-self-update`.

**Comment by MrZammler** [comment]
> > @Ferroin should we get a new nightly version up there, and also check the updater from this PR? Running it now identifies correctly that we are on the latest (-93) do you want to test it?
> 
> That’s probably a good idea, though it means installing now with auto-updates disabled, waiting 24 hours, and then running the updater script from this PR by hand with `--no-updater-self-update`.

All right, I've removed `netdata-updater` from my pi's cron. Let me know I can do a fast run of this PR's netdata-updater when you tell me.

**Comment by Ferroin** [comment]
> > > @Ferroin should we get a new nightly version up there, and also check the updater from this PR? Running it now identifies correctly that we are on the latest (-93) do you want to test it?
> > 
> > 
> > That’s probably a good idea, though it means installing now with auto-updates disabled, waiting 24 hours, and then running the updater script from this PR by hand with `--no-updater-self-update`.
> 
> All right, I've removed `netdata-updater` from my pi's cron. Let me know I can do a fast run of this PR's netdata-updater when you tell me.

Simply waiting until after about 03:00 UTC tomorrow should be sufficient to ensure that there’s a pending update available.

**Comment by tkatsoulas** [comment]
Double down the install and update process on Ubuntu 18.04/20.4/22.04 and centOS 7. The kickstart script from this PR fetches and installs static (when you parse the option `--static-only`) version from GitHub releases (v1.37.0-93-nightly) I am not so sure about that but I think the newly installed netdata-updater script under `/opt/netdata/usr/libexec/netdata/netdata-updater.sh` should observe the new release channel. At the moment it tries to update from the current old channel (GCS) and updates to the Agent to the version (v1.37.0-115-gdb0eb4556)

To sum up, the expected behaviour TTBOMK is the `netdata-updater`  script to just says that the Agent is up-to-date (since the last version is the v1.37.0-93-nightly in the GH releases)

**Comment by Ferroin** [comment]
> Doubling down the install and update process on Ubuntu 18.04/20.04/22.04 and centOS 7. The kickstart script from this PR fetches and installs static (when you parse the option `--static-only`) version from GitHub releases (v1.37.0-93-nightly) I am not so sure about that but I think the newly installed netdata-updater script under `/opt/netdata/usr/libexec/netdata/netdata-updater.sh` should observe the new release channel. At the moment it tries to update from the current old channel (GCS) and updates the Agent to the version (v1.37.0-115-gdb0eb4556)

You need to also pass `--no-updater-self-update` to get that behavior at the moment, because the updater script will update itself to the copy in the master branch on GitHub before trunning updates (this is actually _really_ important for recovering from certain types of bugs in the updater).
 
> To sum up, the expected behavior TTBOMK is the `netdata-updater` script to just says that the Agent is up-to-date (since the latest version is the v1.37.0-93-nightly in the GH releases)

The GH releases are actually out of date, though I’m not sure way at the moment, looking into this now.

**Comment by Ferroin** [comment]
> > To sum up, the expected behavior TTBOMK is the `netdata-updater` script to just says that the Agent is up-to-date (since the latest version is the v1.37.0-93-nightly in the GH releases)
> 
> The GH releases are actually out of date, though I’m not sure way at the moment, looking into this now.

Correction, there’s an issue with publishing of nightly releases because of how GitHub decides to sort releases. Working on a fix for that.

**Comment by Ferroin** [comment]
> > > To sum up, the expected behavior TTBOMK is the `netdata-updater` script to just says that the Agent is up-to-date (since the latest version is the v1.37.0-93-nightly in the GH releases)
> > 
> > 
> > The GH releases are actually out of date, though I’m not sure way at the moment, looking into this now.
> 
> Correction, there’s an issue with publishing of nightly releases because of how GitHub decides to sort releases. Working on a fix for that.

OK, this should be fixed starting with tonight’s nightlies.

**Comment by MrZammler** [comment]
Ok, checked the updater script from this PR, correctly updated the -93 to -140 version.

**Review by MrZammler — APPROVED** [review]
